homeassistant/components/hue/migration.py

```diff
@@ -0,0 +1,178 @@
+            # the device will now have both the old and the new identifier
+            identifiers = {(DOMAIN, hue_dev.id), (DOMAIN, mac)}
+            hass_dev = dev_reg.async_get_or_create(
+                config_entry_id=entry.entry_id, identifiers=identifiers
+            )
+            LOGGER.info("Migrated device %s (%s)", hass_dev.name, hass_dev.id)
+            # loop through al entities for device and find match
+            for ent in async_entries_for_device(ent_reg, hass_dev.id, True):
+                # migrate light
+                if ent.entity_id.startswith("light"):
+                    # should always return one lightid here
+                    new_unique_id = next(iter(hue_dev.lights))
+                    if ent.unique_id == new_unique_id:
+                        continue  # just in case
+                    LOGGER.info(
+                        "Migrating %s from unique id %s to %s",
+                        ent.entity_id,
+                        ent.unique_id,
+                        new_unique_id,
+                    )
+                    ent_reg.async_update_entity(
+                        ent.entity_id, new_unique_id=new_unique_id
+                    )
+                    continue
+                # migrate sensors
+                matched_dev_class = sensor_class_mapping.get(
+                    ent.device_class or "unknown"
+                )
+                if matched_dev_class is None:
+                    # this may happen if we're looking at orphaned or unsupported entity
+                    LOGGER.warning(
+                        "Skip migration of %s because it no longer exists on the bridge",
+                        ent.entity_id,
+                    )
+                    continue
+                for sensor in api.devices.get_sensors(hue_dev.id):
+                    if sensor.type != matched_dev_class:
+                        continue
+                    new_unique_id = sensor.id
+                    if ent.unique_id == new_unique_id:
+                        break  # just in case
+                    LOGGER.info(
+                        "Migrating %s from unique id %s to %s",
+                        ent.entity_id,
+                        ent.unique_id,
+                        new_unique_id,
+                    )
+                    ent_reg.async_update_entity(ent.entity_id, new_unique_id=sensor.id)
+                    break
+
+        # migrate entities that are not connected to a device (groups)
+        for ent in entities_for_config_entry(ent_reg, entry.entry_id):
+            if ent.device_id is not None:
+                continue
+            if ent.unique_id is None:
+                continue  # not sure if this is possible at all ?
+            v1_id = f"/groups/{ent.unique_id}"
+            hue_group = api.groups.room.get_by_v1_id(
+                v1_id
+            ) or api.groups.zone.get_by_v1_id(v1_id)
```

Comment:
We only support rooms for groups:

https://github.com/home-assistant/core/blob/00b1c2bb70a781241f8a68b9429e4e1dccf0c379/homeassistant/components/hue/light.py#L313-L320

Reply:
ah okay that explains it. So the old implementation only created grouped lights for rooms and not for zones, right.

Reply:
fixed